ui-service/src/main/frontend/hotel-booking-ui/src/components/search-components/RoomSelectorComponent.tsx

```diff
@@ -0,0 +1,161 @@
+    const handleChange = (key: 'rooms' | 'adults' | 'children', increment: boolean) => {
+        const currentValue = key === 'rooms' ? rooms : key === 'adults' ? adults : children;
+        const newValue = increment ? currentValue + 1 : currentValue - 1;
+        onRoomDataChange(key, newValue);
+    };
+
+    // Render counter controls for rooms, adults, and children
+    const renderCounter = (label: string, key: 'rooms' | 'adults' | 'children') => (
+        <div className="hotel-search-counter">
+            <span className="hotel-search-label">{label}</span>
+            <div className="hotel-search-controls">
+                <button
+                    type="button"
+                    onClick={() => handleChange(key, false)}
+                    className="hotel-search-control-button"
+                    disabled={
+                        (key === 'rooms' && rooms <= 1) ||
+                        (key === 'adults' && adults <= 1) ||
+                        (key === 'children' && children <= 0)
+                    }
+                >
+                    <Minus size={16}/>
+                </button>
+                <span>{key === 'rooms' ? rooms : key === 'adults' ? adults : children}</span>
+                <button
+                    type="button"
+                    onClick={() => handleChange(key, true)}
+                    className="hotel-search-control-button"
+                    disabled={
+                        (key === 'rooms' && rooms === 9) ||
+                        (key === 'adults' && adults === 5) ||
+                        (key === 'children' && children === 5)
+                    }
+                >
+                    <Plus size={16}/>
+                </button>
+            </div>
+        </div>
+    );
+
+    const renderGroupBookingNotice = () => {
+        if (rooms >= 9) {
+            return (
+                <div className="group-notice">
+                    <div className="group-notice-content">
+                        <BedDouble className="notice-icon"/>
+                        <div className="notice-text">
+                            <p>
+                                Need more rooms? Special group rates available with our hotel partner, hotelplanner.com
+                            </p>
+                            <a href="#" className="notice-link">
+                                10 Rooms or More
+                                <ChevronDown className="notice-arrow"/>
+                            </a>
+                        </div>
+                    </div>
+                </div>
+            );
+        }
+        return null;
+    };
+
+    const renderAddChildren = () => {
+        if (children === 0) return null;
+
+        const getAgeText = (age: number) => {
+            if (age === 0) return 'Infant';
+            return `${age} ${age === 1 ? 'yr' : 'yrs'}`;
+        };
+
+        return Array.from({ length: children }).map((_, index) => (
```

Comment:
Does the search form data type need to be updated to include children ages so that a new event handler can be added here to save age selection?

Reply:
No done yet

Reply:
Updated, please pull the latest code. 
![image](https://github.com/user-attachments/assets/61bc99f8-a8cd-4e40-93df-cd1c01e015c2)